handler/api/src/main/java/io/knotx/fragments/handler/api/metadata/NodeMetadata.java

```diff
@@ -78,6 +92,44 @@ public OperationMetadata getOperation() {
     return operation;
   }
 
+  public void markProcessingStart() {
+    processingStartTimestamp = System.currentTimeMillis();
+  }
+
+  public void markProcessingEnd() {
+    processingEndTimestamp = System.currentTimeMillis();
+  }
+
+  public void calculateTimestampsBasedOnSubtasks(Map<String, NodeMetadata> nodes) {
+    if (type == NodeType.COMPOSITE) {
+      List<NodeMetadata> nestedMetadata = nestedNodes.stream()
+          .map(nodes::get)
+          .collect(Collectors.toList());
+
+      nestedMetadata.forEach(nodeMetadata -> nodeMetadata.calculateTimestampsBasedOnSubtasks(nodes));
+
+      processingStartTimestamp = nestedMetadata.stream()
+          .map(NodeMetadata::getProcessingStartTimestamp)
+          .min(Comparator.naturalOrder())
```

Comment:
What does this Comparator do?

Reply:
Just a standard comparison where 1 < 2 < 3 etc

Reply:
I'd go for `mapToLong(...).min()`, just because it does not require thinking about the ordering. But it's a matter of personal taste I guess 😄